src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -888,15 +890,17 @@ private void WriteObject(object instance, IObjectWritable? instanceAsWritable)
 
                 if (_recursionDepth % MaxRecursionDepth == 0)
                 {
+                    _cancellationToken.ThrowIfCancellationRequested();
+
                     // If we're recursing too deep, move the work to another thread to do so we
-                    // don't blow the stack.  'LongRunning' ensures that we get a dedicated thread
-                    // to do this work.  That way we don't end up blocking the threadpool.
-                    var task = Task.Factory.StartNew(
-                        obj => WriteObjectWorker((IObjectWritable)obj!),
-                        writable,
-                        _cancellationToken,
-                        TaskCreationOptions.LongRunning,
-                        TaskScheduler.Default);
+                    // don't blow the stack.
```

Comment:
The original comment mentions that we want to avoid blocking the threadpool. It feels like we are adding the 'SerializationThreadPool' type because we don't have a built-in alternative that allows us to pool threads while also not risking blocking other kinds of work in the "shared" threadpool. Is that correct?